packages/core/src/shared/lsp/utils/platform.ts

```diff
@@ -26,12 +26,25 @@ function getEncryptionInit(key: Buffer): string {
     return JSON.stringify(request) + '\n'
 }
 
+// use dynamic linking to let node binary
+// pick up GLIBC >= 2.28
+const al2Config = {
+    path: '/opt/vsc-sysroot/lib64/ld-linux-x86-64.so.2',
+    args: ['--library-path', '/opt/vsc-sysroot/lib64'],
+}
+function createNodeProcess(nodePath: string, args: string[]) {
+    return new ChildProcess(
+        isAmazonInternalOs() ? al2Config.path : nodePath,
+        isAmazonInternalOs() ? [...al2Config.args, nodePath, ...args] : [...args],
+        { logging: 'no' }
+    )
+}
 /**
  * Checks that we can actually run the `node` executable and execute code with it.
  */
 export async function validateNodeExe(nodePath: string, lsp: string, args: string[], logger: Logger) {
     // Check that we can start `node` by itself.
-    const proc = new ChildProcess(nodePath, ['-e', 'console.log("ok " + process.version)'], { logging: 'no' })
+    const proc = createNodeProcess(nodePath, ['-e', 'console.log("ok")'])
```

Comment:
any reason why we had to drop the process.version? it might help give us a hint on if the correct node version was picked up

Reply:
my miss!